### DISCUSSION TITLE
Intended way for states in short living workflow

### DISCUSSION THREAD

**Discussion by BoasE** [elsa-workflows/elsa-core#2667] [Q&A]
I have a short living workflow. Basically it gets a type with a list of tasks and the workflow should rearrange the tasks in some given ways.
How should this type exist around the activities? Current. I'm working with the ```WorkflowExecutionContext.Input```
for exampler

```
  if (context.WorkflowExecutionContext.Input is Worksheet wks)
        {
            wks.CurrentTask= wks.ImportTasks.First();
        }
```

Is this the intended way?  or should I think about context providers?

**Comment by sfmskywalker** [comment]
I think your example looks appropriate.
It's good to have custom activities that perform a specific task.

Context providers are useful if your workflow revolves around a single type and you want to prevent having multiple activities be responsible for loading/saving the domain model.
